src/btree/bt_cursor.c

```diff
@@ -325,11 +326,20 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint64_t recno, bool *vali
         upd = (page->modify != NULL && page->modify->mod_row_update != NULL) ?
           page->modify->mod_row_update[cbt->slot] :
           NULL;
+
+        if (key == NULL) {
+            WT_RET(__wt_scr_alloc(session, 0, &unpack_key));
+            WT_RET(__wt_row_leaf_key(
+              session, cbt->ref->page, &cbt->ref->page->pg_row[cbt->slot], unpack_key, true));
+            key = unpack_key;
+        }
```

Comment:
Can we delay this to the time we actually need to search the history store.

Reply:
Yes, that is a good idea. Will do.

Reply:
Fixed.